[chip] Add outlined variant

orporan opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->
Added support for outlined chip as demonstrated in [material design guidelines](https://material.io/design/components/chips.html#input-chips)

orporan committed:
add support for outlined chip

orporan committed:
fix ChipsPlayground

orporan committed:
yarn prettier

orporan committed:
yarn docs:api

orporan committed:
update size limit

orporan committed:
Add hover and focus background color for outlined chip

orporan committed:
yarn prettier

orporan committed:
Add focus state for deletable outlined chip

orporan committed:
Fix chip documentation issue

orporan committed:
Revert last commit

mbrookes commented:
The style descriptions are likely failing because they are split over two lines - at the moment that isn't supported. If the descriptions can't be condensed to 100 chars, we'll have to fix that.

orporan commented:
I'm talking about this error:

> error An unexpected error occurred: "https://registry.yarnpkg.com/builtin-modules/-/builtin-modules-1.1.1.tgz: Request failed \"500 Internal Server Error\"".

orporan committed:
empty commit to trigger build

orporan commented:
@mbrookes
I got a failure in argos because I changed the Demo page to show the new Chip behavior.
[https://www.argos-ci.com/mui-org/material-ui/builds/31206](https://www.argos-ci.com/mui-org/material-ui/builds/31206) 
What do I do about it?

mbrookes commented:
I have approved the visual diff, but should outlined perhaps have its own demo section?

orporan committed:
Add demo seperate demo section to outlined chip

orporan committed:
empty commit to trigger build

orporan committed:
empty commit to trigger build

orporan commented:
@mbrookes 
Done :)

oliviertassinari commented on docs/src/pages/demos/chips/chips.md:
```diff
@@ -25,6 +25,13 @@ and (string) Avatar.
 
+### Outlined Chips
+
+Outlined chips offer alternative style 
```
Period?

mbrookes replied:
"offer alternative style" -> "offer an alternative style."

oliviertassinari commented on docs/src/pages/demos/chips/OutlinedChips.js:
```diff
@@ -0,0 +1,112 @@
+  alert('You clicked the Chip.'); // eslint-disable-line no-alert
+}
+
+function Chips(props) {
```
`OutlinedChips`

oliviertassinari commented on packages/material-ui/src/Chip/Chip.d.ts:
```diff
@@ -29,7 +30,9 @@ export type ChipClassKey =
   | 'deleteIcon'
   | 'deleteIconPrimary'
-  | 'deleteIconSecondary';
+  | 'deleteIconSecondary'
```
The PR adds 5 new classes. There is only two here.

orporan replied:
I just noticed that for deleteIconPrimary and deleteIconSecondary the .d.ts file doesn't match the implementation (deleteIconColorPrimary, deleteIconColorSecondary).
what should I do about it?

oliviertassinari commented on packages/material-ui/src/Chip/Chip.js:
```diff
@@ -146,20 +174,34 @@ export const styles = theme => {
       },
     },
+    /* Styles applied to the deleteIcon element if `color="primary"` and `variant="outlined"`. */
+    deleteIconColorPrimaryOutlined: {
```
We should probably reverse *color primary* and *outlined*.

oliviertassinari reviewed (commented):
A quick review while I'm at the beach 🌊☀️.

orporan committed:
PR review fixes

orporan committed:
PR review fixes

orporan committed:
empty commit to trigger build

orporan committed:
empty commit to trigger build

orporan commented:
@mbrookes @oliviertassinari Thanks for the quick review. I addressed all comments and the tests are passing (except for argos, but that's due to the intentional change in Demo components)

mbrookes committed:
Merge branch 'master' into add-outlined-chip

orporan committed:
empty commit to trigger build

orporan committed:
fix size limit after merge

orporan commented:
@mbrookes Thank you for merging the master branch into the PR. I adjusted the package size accordingly and all the tests are passing now.
I hope its okay that I'm asking, but how soon can this be released? I need to use it for something :)

mbrookes reviewed (approved):
@orporan Looks good to go. 👌 

Once merged, it will be released in the next weekly release cycle.

orporan commented:
@mbrookes great!
Is there anything else I need to do for this to get merged?

mbrookes commented:
@orporan Sorry, yes - just realized we don't have any tests for the new variant. It should be easy enough to duplicate an existing test or two to make sure the correct classes are applied when the variant is set to outlined.

mbrookes reviewed (changes requested):
@orporan Sorry, yes. I just realised we don't have any tests for the new variant. It should be straightforward enough to duplicate a test and modify it to check that the correct classes are applied when the outlined variant is specified.

orporan committed:
Add tests for outlined chip

orporan committed:
yarn prettier

mbrookes commented on docs/src/pages/demos/chips/OutlinedChips.js:
```diff
@@ -0,0 +1,112 @@
+            <FaceIcon />
+          </Avatar>
+        }
+        label="Clickable Link Chip"
```
Not your fault as it came from the prior demo, but the labels on the last three chips don't seem to match their purpose. The last two are primary color and secondary color, and the third-last seems to be a smoosh of a whole bunch - primary clickable custom delete.

Please could you give them appropriate labels in both examples?

mbrookes reviewed (commented):
That was quick. 😄 One last request...

orporan committed:
empty commit to trigger build

orporan committed:
Fix demo labels for chip

orporan commented:
@mbrookes all done :)

mbrookes commented:
👌

oliviertassinari committed:
Update chips.md

oliviertassinari committed:
let's merge

mbrookes commented:
@oliviertassinari Ever the eagle eye! 👍

mbrookes commented:
@orporan Thanks for your perseverance! Less than a week 'till it's released.

oliviertassinari commented:
@mbrookes I'm releasing a v3.0.1 with the peer dependency fix.

orporan commented:
@mbrookes @oliviertassinari Thanks! it was a pleasure :)
